### PR TITLE
Dark theme changes

### PR DESCRIPTION
**Pull request by UsamaSaleem14**
Work in this PR:

1. Added tailwind configuration for dark theme
2. Added a button to toggle between light and dark theme

### CODE REVIEW

**Comment by TBomberman** [comment]
image
1. Make sides also dark.
2. Remove package.lock.
3. Check both windows and class strategy.

File: src/App.tsx [hunk 1]
```diff
@@ -7,14 +7,109 @@ import Header from "./components/Header";
...
+      }
+    }
+  }, []);
+
+  const changeTheme = (theme: string) => {
+    if (theme === "light") {
+      setIsDark(false);
+      localStorage.removeItem("theme");
+      document.documentElement.classList.remove("dark");
+      const elem = document.getElementById("sunDiv");
```
Comment (TBomberman) [inline src/App.tsx]: use jsx and one state variable

File: src/App.tsx [hunk 2]
```diff
@@ -7,14 +7,109 @@ import Header from "./components/Header";
...
+        elem.classList.add("rounded-3xl");
+        elem.classList.add("text-white");
+      }
+    } else {
+      setIsDark(false);
+      localStorage.removeItem("theme");
+      document.documentElement.classList.remove("dark");
+      const elem = document.getElementById("sunDiv");
+      if (elem !== null) {
+        elem.classList.add("bg-white");
```
Comment (TBomberman) [inline src/App.tsx]: Put this into tailwinds style.

File: src/App.tsx [hunk 3]
```diff
@@ -7,14 +7,109 @@ import Header from "./components/Header";
...
+      if (elem2 !== null) {
+        elem2.classList.remove("bg-white");
+        elem2.classList.remove("rounded-3xl");
+        elem2.classList.remove("text-black");
+      }
+    }
   };
   return (
     <div className={style.wrapper}>
+      <div className={style.themeContainer}>
```
Comment (TBomberman) [inline src/App.tsx]: Can this be a component?

File: src/App.tsx [hunk 4]
```diff
@@ -7,14 +7,109 @@ import Header from "./components/Header";
...
+  }, []);
+
+  const changeTheme = (theme: string) => {
+    if (theme === "light") {
+      setIsDark(false);
+      localStorage.removeItem("theme");
+      document.documentElement.classList.remove("dark");
+      const elem = document.getElementById("sunDiv");
+      if (elem !== null) {
+        elem.classList.add("bg-white");
```
Comment (TBomberman) [inline src/App.tsx]: These class list should be tailwinds class in the toggle component.

File: src/App.tsx [hunk 5]
```diff
@@ -7,14 +7,109 @@ import Header from "./components/Header";
...
+      "w-full flex flex-row justify-center max-w-screen-2xl px-[200px] dark:bg-dark-background min-h-screen items-start",
     container: "flex flex-col justify-center mt-6",
+    themeIcon: "text-xl m-2",
+    themeIconMoonLight: "text-xl m-2 text-light-font-lightV2",
+    themeContainer:
+      "fixed bottom-2 right-2 flex justify-between items-center bg-light-primary-grey dark:bg-white py-1 px-2 rounded-[30px] z-50",
+    iconDiv: "cursor-pointer",
+  };
+
+  const [isDark, setIsDark] = useState(false);
```
Comment (TBomberman) [inline src/App.tsx]: This variable is not needed, can check the html class if it's dark or not.

File: src/App.tsx (hunk 1 above)
Comment (UsamaSaleem14) [reply src/App.tsx]: Removed this state variable as we do not need it

File: src/App.tsx (hunk 2 above)
Comment (UsamaSaleem14) [reply src/App.tsx]: Done

File: src/App.tsx (hunk 3 above)
Comment (UsamaSaleem14) [reply src/App.tsx]: Created a separate component for theme switching

File: src/App.tsx (hunk 4 above)
Comment (UsamaSaleem14) [reply src/App.tsx]: Done

File: src/App.tsx (hunk 5 above)
Comment (UsamaSaleem14) [reply src/App.tsx]: Done

**Comment by UsamaSaleem14** [comment]
> image
> 
> 1. Make sides also dark.
> 2. Remove package.lock.
> 3. Check both windows and class strategy.

Done

**Review by TBomberman — APPROVED** [review]

**Pull request merged**